celery/backends/elasticsearch.py

```diff
@@ -88,6 +89,7 @@ def set(self, key, value):
             self._index(key, data, refresh=True)
 
     def _index(self, id, body, **kwargs):
+        body = dict((string(k), v) for k, v in body.items())
         return self.server.index(
```

Comment:
Also, why isn't this a dictionary comprehension? We don't support Python 2.6 anymore.